File: cs/Problems/Day3Test.cs
namespace aoc24.Problems;

public class Day3Test
{
    private readonly Day3 sut = new();

    [Theory, InlineData(161)]
    public void TestSet_ShouldYield_Result(int expected)
    {
        var input = InputReader.ReadProblemInput("day3_1");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }

    [Theory, InlineData(188741603)]
    public void FullSet_ShouldYield_Result(int expected)
    {
        var input = InputReader.ReadProblemInput("day3_2");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }
}

File: go/problems/day3_test.go
package problems

import "testing"

func TestDay3_TestSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day3{}, "day3_1", 161)
}

func TestDay3_FullSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day3{}, "day3_2", 188741603)
}
